inspirehep/dojson/utils.py

```diff
@@ -131,15 +131,27 @@ def remove_duplicates_from_list_of_dicts(ld):
     """Remove duplicates from a list of dictionaries preserving the order.
 
     We can't use the generic list helper because a dictionary isn't
-    hashable. Taken from http://stackoverflow.com/a/9427216/374865."""
+    hashable. Adapted from http://stackoverflow.com/a/9427216/374865."""
+
+    def _freeze(o):
+        """Recursively freezes a dict in an hashable object.
+
+        Adapted from http://stackoverflow.com/a/21614155/374865."""
+        if isinstance(o, dict):
+            return frozenset({k: _freeze(v) for k, v in o.items()}.items())
+        elif isinstance(o, (list, tuple)):
+            return tuple([_freeze(v) for v in o])
+        else:
+            return o
```

Comment:
BTW Just confirming that validation is happening after signals `before_record_insert/update` are sent. So we can call your `remove_duplicates_from_list_of_dicts()` there, and remove all duplicate stripping from dojson.

Additionally I would move there also the calls to `strip_empty_values()` in https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/migrator/tasks.py#L368-L382